docs/source/share_dataset.rst

```diff
@@ -220,29 +220,48 @@ Other files can safely be deleted.
 Uploading your files
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Once the repo is cloned, you can add the dataset script and dataset infos.
+Once the repo is cloned, If you need to add data files, instead of relying on the data to be hosted
+elsewhere, add these files using the following steps. Let's say that the files you're adding are ``*.json`` files, then:
 
-You can add these files to the staging environment and verify that they have been correctly staged with the ``git
-status`` command:
+.. code-block:: bash
+
+    cp /somewhere/data/*.json .
+    git lfs track *.json
+    git add .gitattributes
+    git commit -m "track json files"
+    git add *.json

```

Comment:
wouldn't you reverse those two lines? if you want to actually commit your json files?

Reply:
Yes indeed. Or you would need a second commit after adding the json files

Let's reverse those two line and have one commit

Reply:
Actually from the git-lfs github page they recommend to commit after each git track command:
https://github.com/git-lfs/git-lfs
> After any invocation of git-lfs-track(1) or git-lfs-untrack(1), you must commit changes to your .gitattributes file.

Reply:
hmm, I don't think they're saying that you have to commit after each change to .gitattributes. Just that you have to commit the file to version control it.

Reply:
.gitattributes is just a text file (like .gitignore etc) there's nothing special about it

Reply:
Yes right.
What do you think about having two commits ? It's one more command but I feel it makes it clear that your first need to track and add your .gitattributes before actually adding your data files

Reply:
I would personally do just one commit, but no strong opinion

I do not believe you need to `add` your .gitattributes before adding your data files (just track)

Reply:
Actually I think you do need to add your .gitattributes first.
Let me check

Reply:
I just did
```
git clone https://huggingface.co/datasets/lhoestq/testt
cd testt
touch dummy.yo
git lfs track *.yo
git add .
git commit -m "test"
git push
```
and dummy.yo doesn't seem to be on lfs
```
❯ curl -I https://huggingface.co/datasets/lhoestq/testt/resolve/main/dummy.yo
HTTP/1.1 200 OK
Server: nginx/1.18.0 (Ubuntu)
Date: Tue, 16 Feb 2021 11:07:28 GMT
Content-Type: application/octet-stream
Content-Length: 0
Connection: keep-alive
X-Powered-By: Express
ETag: "e69de29bb2d1d6434b8b29ae775ad8c2e48c5391"
```

Reply:
there might be a special case for empty files (their sha256 is... non-existent)

Can you try the same thing with a non-empty file?

Reply:
Indeed it worked for an non-empty file:
```
echo "Hello world" > dummy.foo
git lfs track *.foo
git add .
git commit -m "test2"
git push
```
```
❯ curl -I https://huggingface.co/datasets/lhoestq/testt/resolve/main/dummy.fooHTTP/1.1 302 Found
Server: nginx/1.18.0 (Ubuntu)
Date: Tue, 16 Feb 2021 11:18:06 GMT
Content-Type: text/plain; charset=utf-8
Content-Length: 140
Connection: keep-alive
X-Powered-By: Express
X-Linked-Size: 12
X-Linked-ETag: "1894a19...183ea2afc7c48f"
Location: https://cdn-lfs.huggingface.co/datasets/lhoestq/testt/1894a19...83ea2afc7c48f
Vary: Accept
```